ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -191,14 +200,42 @@ private static SourceDescriptionWithWarnings describeSource(
       ), statementText);
     }
 
+    List<RunningQuery> sourceQueries = getQueries(ksqlEngine,
+        q -> q.getSourceNames().contains(dataSource.getName()));
+    List<RunningQuery> sinkQueries = getQueries(ksqlEngine,
+        q -> q.getSinkName().equals(dataSource.getName()));
+
     Optional<org.apache.kafka.clients.admin.TopicDescription> topicDescription = Optional.empty();
+    Optional<ConsumerGroupDescription> consumerGroupDescription = Optional.empty();
+    Map<TopicPartition, OffsetAndMetadata> topicAndConsumerOffsets = new LinkedHashMap<>();
+    Map<TopicPartition, ListOffsetsResultInfo> topicAndStartOffsets = new LinkedHashMap<>();
+    Map<TopicPartition, ListOffsetsResultInfo> topicAndEndOffsets = new LinkedHashMap<>();
     final List<KsqlWarning> warnings = new LinkedList<>();
     if (extended) {
       try {
         topicDescription = Optional.of(
             serviceContext.getTopicClient().describeTopic(dataSource.getKafkaTopicName())
         );
-      } catch (final KafkaException | KafkaResponseGetFailedException e) {
+        String serviceId = "default"; //FIXME not sure how to get this
```

Comment:
Not sure how to get this from config.

Reply:
I think you're looking for `KsqlConfig.KSQL_SERVICE_ID_CONFIG` the config is availabile in the `ConfigureStatement` class